src/EditorFeatures/Core.Wpf/BackgroundWorkIndicator/BackgroundWorkIndicatorContext.cs

```diff
@@ -21,281 +21,54 @@ namespace Microsoft.CodeAnalysis.Editor.BackgroundWorkIndicator;
-
-                // Ensure we only dismiss once.
-                if (_dismissed)
-                    return;
-
-                _dismissed = true;
-
-                // Unhook any event handlers we've setup.
-                //
-                // note we have to disconnect from dismissal notifications so our own dismiss below doesn't cause us to
-                // re-enter and cancel ourselves.
-                _toolTipPresenter.Dismissed -= OnToolTipPresenterDismissed;
-                _subjectBuffer.Changed -= OnTextBufferChanged;
-                _textView.LostAggregateFocus -= OnTextViewLostAggregateFocus;
-
-                // Finally, dismiss the actual tool-tip.
-                _toolTipPresenter.Dismiss();
-
-                // Let our factory know that we were disposed so it can let go of us as well.
-                _factory.OnContextDisposed(this);
-            }
-
-            async ValueTask UpdateUIAsync()
-            {
-                // Build the current description in the background, then switch to the UI thread to actually update the
-                // tool-tip with it.
-                var data = this.BuildData();
-
-                await this.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-
-                // If we've been dismissed already, then no point in continuing.
-                if (_dismissed)
-                    return;
-
-                // Todo: build a richer tool-tip that makes use of things like the progress reported, and perhaps has a
-                // close button.
-                _toolTipPresenter.StartOrUpdate(
-                    _trackingSpan, new[] { string.Format(EditorFeaturesResources._0_Esc_to_cancel, data.description) });
-            }
+            _indicator.Dispose();
+            OnDisposed?.Invoke(this, null);
         }
 
         public IUIThreadOperationScope AddScope(bool allowCancellation, string description)
         {
-            var scope = new BackgroundWorkIndicatorScope(this, description);
-            lock (this.Gate)
-            {
-                _scopes = _scopes.Add(scope);
-            }
-
-            // We changed.  Enqueue work to make sure the UI reflects this.
-            this.EnqueueUIUpdate();
+            var backgroundScope = _indicator.AddScope(description);
+            var scope = new BackgroundWorkIndicatorScope(backgroundScope, description, this);
+            _scopes.Add(scope);
             return scope;
         }
 
-        public void RemoveScope(BackgroundWorkIndicatorScope scope)
-        {
-            lock (this.Gate)
-            {
-                Contract.ThrowIfFalse(_scopes.Contains(scope));
-                _scopes = _scopes.Remove(scope);
-            }
-
-            // We changed.  Enqueue work to make sure the UI reflects this.
-            this.EnqueueUIUpdate();
-        }
-
-        private (string description, ProgressInfo progressInfo) BuildData()
+        public void TakeOwnership()
```

Comment:
do we use this?

Reply:
Vestigial from us merging with `IUIThreadOperationScope` even though the platform doesn't. This is a rough edge we have to deal with if we want complete abstraction. I'm happy to not have out own wrappers and move to just the platform implementation